packages/expo-updates/android/src/main/java/expo/modules/updates/loader/RemoteLoader.kt

```diff
@@ -55,9 +55,12 @@ class RemoteLoader internal constructor(
     assetEntity: AssetEntity,
     updatesDirectory: File?,
     configuration: UpdatesConfiguration,
+    requestedUpdate: UpdateEntity?,
     callback: AssetDownloadCallback
   ) {
-    mFileDownloader.downloadAsset(assetEntity, updatesDirectory, callback)
+    val embeddedUpdate = loaderFiles.readEmbeddedUpdate(context, configuration)?.updateEntity
+    val extraHeaders = FileDownloader.getExtraHeadersForRemoteAssetRequest(assetEntity, launchedUpdate, embeddedUpdate, requestedUpdate)
```

Comment:
not ideal to re-compute the extra header for each asset downloading. maybe we can compute once and pass the computed result to this function.

Reply:
I moved the `embeddedUpdate` to the `downloadAllAssets` function in the `Loader` class but we can't really move the `extraHeaders` variable there because the base `Loader` class does not have access to `launchedUpdate`